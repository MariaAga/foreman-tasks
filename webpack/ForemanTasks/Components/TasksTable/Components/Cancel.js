import React from 'react';
import { Button } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

export const Cancel = ({ id, cancellable }) => (
  <Button
    bsSize="small"
    data-method="post"
    href={`/foreman_tasks/tasks/${id}/cancel`}
    disabled={!cancellable}
  >
    {__('Cancel')}
  </Button>
);

Cancel.propTypes = {
  id: PropTypes.string.isRequired,
  cancellable: PropTypes.bool.isRequired,
};
