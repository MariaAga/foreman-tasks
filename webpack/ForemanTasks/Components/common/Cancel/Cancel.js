import React from 'react';
import { Button } from 'patternfly-react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';

export const Cancel = ({ id, name, cancellable, cancelTaskAction }) => (
  <Button
    bsSize="small"
    onClick={() => cancelTaskAction(id, name)}
    disabled={!cancellable}
  >
    {__('Cancel')}
  </Button>
);

Cancel.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  cancellable: PropTypes.bool.isRequired,
  cancelTaskAction: PropTypes.func.isRequired,
};
