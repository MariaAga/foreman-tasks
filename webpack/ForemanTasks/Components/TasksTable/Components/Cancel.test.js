import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import { Cancel } from './Cancel';

const fixtures = {
  'render with cancellable true props': { id: 'id', cancellable: true },
  'render with cancellable false props': { id: 'id', cancellable: false },
};

describe('Cancel', () => testComponentSnapshotsWithFixtures(Cancel, fixtures));
