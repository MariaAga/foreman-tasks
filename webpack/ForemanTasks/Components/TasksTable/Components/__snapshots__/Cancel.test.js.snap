// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cancel render with cancellable false props 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsSize="small"
  bsStyle="default"
  data-method="post"
  disabled={true}
  href="/foreman_tasks/tasks/id/cancel"
>
  Cancel
</Button>
`;

exports[`Cancel render with cancellable true props 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsSize="small"
  bsStyle="default"
  data-method="post"
  disabled={false}
  href="/foreman_tasks/tasks/id/cancel"
>
  Cancel
</Button>
`;
